### PR TITLE
doc: add a Working with nRF52 user guide

### PR DESCRIPTION
**Pull request by ru-fu**
Add a user guide that gives information about how to work
with nRF52 Series devices.

PDF output: 
[Working with nRF52 — nRF Connect SDK 1.2.pdf](https://github.com/NordicPlayground/fw-nrfconnect-nrf/files/4288425/Working.with.nRF52.nRF.Connect.SDK.1.2.pdf)

Signed-off-by: Ruth Fuchss <ruth.fuchss@nordicsemi.no>

### CODE REVIEW

File: doc/nrf/ug_nrf52.rst [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+When you develop a Bluetooth Low Energy (LE) application, you must use the Bluetooth LE stack.
+This stack is split into two different layers: the Bluetooth LE Host and the Bluetooth LE Controller.
+
+The |NCS| uses Zephyr's implementation of the Bluetooth LE Host.
+For the Bluetooth LE Controller, you can choose between the nRF Bluetooth LE Controller that is implemented by Nordic Semiconductor and Zephyr's open-source Bluetooth LE Controller.
+The :ref:`ug_ble_controller` user guide contains more information about the two available controllers and instructions for switching between them.
+
+See :ref:`zephyr:bluetooth` for documentation on Zephyr's Bluetooth LE stack, including both host and controller.
+For documentation about the nRF Bluetooth LE Controller and information on what variants of the controller support with chips, see :ref:`nrfxlib:ble_controller`.
```
Comment (grochu) [inline doc/nrf/ug_nrf52.rst]:
```suggestion
For documentation about the nRF Bluetooth LE Controller and information on what variants of the controller support which chips, see :ref:`nrfxlib:ble_controller`.
```

File: doc/nrf/ug_nrf52.rst [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+   :start-after: esb_intro_start
+   :end-before: esb_intro_end
+
+See the :ref:`ug_esb` user guide for information about how to work with Enhanced ShockBurst.
+To start developing, check out the :ref:`esb_prx_ptx` sample.
+
+Near field communication
+========================
+
+Near field communication (NFC) is a technology for wireless transfer of small amounts of data between two devices.
```
Comment (grochu) [inline doc/nrf/ug_nrf52.rst]:
```suggestion
Near Field Communication (NFC) is a technology for wireless transfer of small amounts of data between two devices that are in close proximity.
```

File: doc/nrf/ug_nrf52.rst [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+It uses very simple communication protocols and therefore allows quicker connection than Bluetooth LE.
+However, because NFC uses magnetic induction to enable communication, the devices must be very close (< 10 cm) to connect.
+
+The stack for developing NFC applications is provided in binary format in the `nrfxlib`_ repository.
```
Comment (grochu) [inline doc/nrf/ug_nrf52.rst]:
```suggestion
|NCS| provides 2 protocol stacks for developing NFC applications: Type 2 Tag and Type 4 Tag. These stacks are provided in binary format in the `nrfxlib`_ repository.
```

File: doc/nrf/ug_nrf52.rst [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+See :ref:`nrfxlib:nfc` for documentation about the NFC stack.
```
Comment (grochu) [inline doc/nrf/ug_nrf52.rst]:
```suggestion
See :ref:`nrfxlib:nfc` for documentation about the NFC stacks.
```

File: doc/nrf/ug_nrf52.rst [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+The main area of application for the nRF52 Series is Bluetooth Low Energy, but you can also use the proprietary ESB protocol, NFC, or USB.
+The following sections give pointers on where to start when working with these protocols.
+
+To test the general capabilities of the 2.4 GHz radio transceiver, use the :ref:`radio_test` sample.
+
+Bluetooth Low Energy
+====================
+
+When you develop a Bluetooth Low Energy (LE) application, you must use the Bluetooth LE stack.
+This stack is split into two different layers: the Bluetooth LE Host and the Bluetooth LE Controller.
```
[the remark this answers is not in the document]
Comment (joerchan) [reply doc/nrf/ug_nrf52.rst]: Correct. There is the Bluetooth Host and Bluetooth LE controller. Also the Host consists of several layers. So the term layer should be used correctly here.

"The Bluetooth software stack is split into two core components: the Bluetooth Host and the Bluetooth LE Controller."

File: doc/nrf/ug_nrf52.rst [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+
+The |NCS| uses Zephyr's implementation of the Bluetooth LE Host.
+For the Bluetooth LE Controller, you can choose between the nRF Bluetooth LE Controller that is implemented by Nordic Semiconductor and Zephyr's open-source Bluetooth LE Controller.
+The :ref:`ug_ble_controller` user guide contains more information about the two available controllers and instructions for switching between them.
```
Comment (jdstapes) [inline doc/nrf/ug_nrf52.rst]: No need to repeat what is in the references.  Suggest replace whole para with:
The nRF Connect SDK Bluetooth stack is fully supported by Nordic for nRF52 series devices.

File: doc/nrf/ug_nrf52.rst [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+
+See :ref:`zephyr:bluetooth` for documentation on Zephyr's Bluetooth LE stack, including both host and controller.
```
Comment (jdstapes) [inline doc/nrf/ug_nrf52.rst]:
```suggestion
See this :ref:`zephyr:bluetooth` for documentation on the Bluetooth LE Host and open source LE Controller.
```

File: doc/nrf/ug_nrf52.rst [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+For documentation about the nRF Bluetooth LE Controller and information on what variants of the controller support with chips, see :ref:`nrfxlib:ble_controller`.
+
+The |NCS| contains a variety of :ref:`ble_samples` that target nRF52 Series devices.
+In addition, you can run any of Zephyr's :ref:`zephyr:bluetooth-samples`.
```
Comment (jdstapes) [inline doc/nrf/ug_nrf52.rst]:
```suggestion
In addition, you can run the ref:`zephyr:bluetooth-samples` included from the Zephyr project.
```

File: doc/nrf/ug_nrf52.rst [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+
+See the :ref:`ug_esb` user guide for information about how to work with Enhanced ShockBurst.
+To start developing, check out the :ref:`esb_prx_ptx` sample.
+
+Near field communication
+========================
+
+Near field communication (NFC) is a technology for wireless transfer of small amounts of data between two devices.
+It uses very simple communication protocols and therefore allows quicker connection than Bluetooth LE.
+However, because NFC uses magnetic induction to enable communication, the devices must be very close (< 10 cm) to connect.
```
Comment (jdstapes) [inline doc/nrf/ug_nrf52.rst]: Not sure on this description.  Suggest replace with:
NFC makes transferring data when devices in close proximity fast and easy.  The range of NFC is typically < 10 cm.

File: doc/nrf/ug_nrf52.rst [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+
+The stack for developing NFC applications is provided in binary format in the `nrfxlib`_ repository.
+See :ref:`nrfxlib:nfc` for documentation about the NFC stack.
+
+The NFC stack requires the NFCT driver for nRF52 devices, which is available as part of `nrfx`_.
```
Comment (jdstapes) [inline doc/nrf/ug_nrf52.rst]: The "nrfx" just links out to git hub.  That gives the impression the user needs to do something to use it.  I would prefer we link to a new page, included at the top level "About the nRF Connect SDK" which explains that nrfx contains Nordic's RTOS independent driver implementations, it is already incorporated in the Zephyr repository and is available separately in github <link>."

File: doc/nrf/ug_nrf52.rst [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+
+See :ref:`nfc_samples` and :ref:`lib_nfc` for lists of samples and libraries that the |NCS| provides.
+
+USB
+===
+
+The |NCS| uses Zephyr's implementation of a USB device stack.
```
Comment (jdstapes) [inline doc/nrf/ug_nrf52.rst]:
```suggestion
The |NCS| contains a USB device stack for the USB 2.0 Full Speed peripheral that is available on a number of the nRF52 devices.
```

File: doc/nrf/ug_nrf52.rst [hunk 12]
```diff
@@ -0,0 +1,95 @@
...
+See :ref:`zephyr:usb_device_stack` for documentation and :ref:`zephyr:usb-samples` for a list of available samples.
+
+The USB stack requires the USBD driver for nRF52 devices, which is available as part of `nrfx`_.
```
Comment (jdstapes) [inline doc/nrf/ug_nrf52.rst]: same comment nrfx

File: doc/nrf/ug_nrf52.rst [hunk 13]
```diff
@@ -0,0 +1,95 @@
+.. _ug_nrf52:
+
+Working with nRF52
+##################
+
+The |NCS| provides support for developing on a variety of nRF52 Series devices, using the respective development kits.
```
Comment (jdstapes) [inline doc/nrf/ug_nrf52.rst]:
```suggestion
The |NCS| provides support for developing on all nRF52 Series devices, and contains board definitions for all development kits and reference design hardware.
```

File: doc/nrf/ug_nrf52.rst [hunk 14]
```diff
@@ -0,0 +1,95 @@
...
+
+Introduction
+************
+
+The nRF52 Series is a family of multiprotocol System on Chip (SoC) devices that embed a powerful yet low-power Arm Cortex-M4 processor.
+All of the nRF52 Series SoCs have support for Bluetooth 5 features, in addition to multiprotocol capabilities (for example, Bluetooth Mesh and 2.4 GHz proprietary).
```
[the remark this answers is not in the document]
Comment (bjorn-tore-taraldsen) [reply doc/nrf/ug_nrf52.rst]: We should not start talking about 15.4 protocols at this point. That has to be added once we have proper support that in NCS

File: doc/nrf/ug_nrf52.rst (hunk 13 above)
[the remark this answers is not in the document]
Comment (ru-fu) [reply doc/nrf/ug_nrf52.rst]: Sentence updated as suggested by Joel.

File: doc/nrf/ug_nrf52.rst (hunk 7 above)
[the remark this answers is not in the document]
Comment (joerchan) [reply doc/nrf/ug_nrf52.rst]: I'm not sure linking to the implementation of the controller is needed. We should only link to documentation describing features support and options.

@cvinayak  @carlescufi Do we have something like that for the zephyr controller?

[the remark this answers is not in the document]
Comment (joerchan) [reply doc/nrf/ug_nrf52.rst]: @ru-fu Right. But we don't use the controller directly, all of the Bluetooth API is through the host.
The thing we need for the controller are how to configure features on/off, or enable/disable feature options.

File: doc/nrf/ug_nrf52.rst (hunk 10 above)
Comment (ru-fu) [reply doc/nrf/ug_nrf52.rst]: Will add a sentence that nrfx is included through the Zephyr repo. The other updates should come as part of including the nrfx documentation.

File: doc/nrf/ug_nrf52.rst (hunk 6 above)
Comment (OisinFitz) [reply doc/nrf/ug_nrf52.rst]: I think the brief explanation that we have in here is nice.  IMO, it's not repetitive, as you need to jump to a link to find more info.

File: doc/nrf/ug_nrf52.rst [hunk 15]
```diff
@@ -0,0 +1,144 @@
...
+The build system uses a combination of device part code and PCA number to identify a board name in the format ``nRF52xxx-PCAxxxxx``.
+For example, for the :ref:`nRF52840 Development Kit (PCA10056) <zephyr:nrf52840_pca10056>`, the board name is ``nRF52840-PCA20056``, and you must use ``nrf52840_pca10056`` as target when building.
+
+Devices in the nRF52 Series are supported by these boards in the Zephyr open source project and in |NCS|.
+Refer to the PCA number for building projects for these boards and find a link to their descriptions.
+
+.. list-table::
+   :header-rows: 1
+
+   * - DK
```
Comment (b-gent) [inline doc/nrf/ug_nrf52.rst]: Suggestions:
- A column 'Device' could make it clearer for new customers which DK is used for which SoC. Also, someone looking for nRF52832 won't find it in the table (only after following the link to Zephyr).
- A note below the table referring to using nRF52840 DK for nRF52811 and nRF52 DK for nRF52810, explaining that the 'bigger' chip is used to emulate the functionality of the 'smaller', similar to what we had in the old SDK: https://infocenter.nordicsemi.com/topic/sdk_nrf5_v16.0.0/nrf52811_user_guide.html

Comment (ru-fu) [reply doc/nrf/ug_nrf52.rst]: Leave as is because the title is supported boards not supported chips.

File: doc/nrf/ug_nrf52.rst [hunk 16]
```diff
@@ -0,0 +1,144 @@
...
+Near Field Communication (NFC) is a technology for wireless transfer of small amounts of data between two devices that are in close proximity.
+It makes transferring data fast and easy when devices are in close proximity.
+The range of NFC is typically <10 cm.
+
+|NCS| provides two protocol stacks for developing NFC applications: Type 2 Tag and Type 4 Tag.
+These stacks are provided in binary format in the `nrfxlib`_ repository.
+See :ref:`nrfxlib:nfc` for documentation about the NFC stacks.
+
+The NFC stack requires the NFCT driver for nRF52 devices, which is available as part of `nrfx`_.
+The nrfx project is included in the |NCS| as a module of the Zephyr repository.
```
Comment (ru-fu) [inline doc/nrf/ug_nrf52.rst]:
```suggestion
The nrfx repository is included in the |NCS| as a module of the Zephyr repository.
```

File: doc/nrf/ug_nrf52.rst (hunk 7 above)
[the remark this answers is not in the document]
Comment (carlescufi) [reply doc/nrf/ug_nrf52.rst]: This is just a todo. At some point we either include technical documentation on the controller's inner workings on that page, or just list its features and an overview of the functionality.

**Pull request merged**